ui/app/components/clients/history.js

```diff
@@ -92,7 +92,45 @@ export default class History extends Component {
   get latestUpgradeData() {
     // {id: '1.9.0', previousVersion: null, timestampInstalled: '2021-11-03T10:23:16Z'}
     // version id is 1.9.0 or earliest upgrade post 1.9.0, timestamp is RFC3339
-    return this.args.model.versionHistory[0] || null;
+    // need to return the last in the array as the data is in ascending order
+    return this.args.model.versionHistory.at(-1) || null;
```

Comment:
Did this change? I thought the `keys` were in descending order, with the latest first, and then in the serializer we map over the keys:
```
 normalizeFindAllResponse(store, primaryModelClass, payload, id, requestType) {
    let normalizedPayload = [];
    payload.keys.forEach((key) => {
      normalizedPayload.push({ id: key, ...payload.key_info[key] });
    });
    return this._super(store, primaryModelClass, normalizedPayload, id, requestType);
  },
  ```

Reply:
☝️ on that note, we only care about the timestamp for whichever key has `previousVersion:null` 

would it make sense to instead have something like : 

``` get findInitialUpgrade () {
return this.args.model.versionHistory.find( version => version.previousVersion === null) || false
 }

Reply:
We do still need some sort of object with version upgrade info because it's passed to the chart to display upgrade info.